### PR TITLE
Update bclconvert: show undetermined barcodes for multiple runs

### PR DESCRIPTION
**Pull request by nvnieuwk**
- [x] Removed the check for the undetermined barcodes plot so the barcodes are also generated when there are multiple runs
- [x] `CHANGELOG.md` has been updated

- [x] There is example tool output for tools in the <https://github.com/ewels/MultiQC_TestData> repository or attached to this PR
- [x] Code is tested and works locally (including with `--lint` flag)
- [x] `docs/README.md` is updated with link to below
- [x] `docs/modulename.md` is created
- [x] Everything that can be represented with a plot instead of a table is a plot
- [x] Report sections have a description and help text (with `self.add_section`)
- [x] There aren't any huge tables with > 6 columns (explain reasoning if so)
- [x] Each table column has a different colour scale to its neighbour, which relates to the data (eg. if high numbers are bad, they're red)
- [x] Module does not do any significant computational work

### CODE REVIEW

**Comment by nvnieuwk** [comment]
ping @matthdsm

**Comment by matthdsm** [comment]
Hi @ewels ,  @andrei-seleznev 

Could you comment as to _why_ mutli lane parsing was disabled? Everything seems to run just fine...
[multiqc_report (1).html.zip](https://github.com/ewels/MultiQC/files/9468701/multiqc_report.1.html.zip)

**Comment by nvnieuwk** [comment]
Fixes #1709

**Comment by alexiswl** [comment]
Hi @matthdsm,  

There's a small discussion on this here: https://github.com/ewels/MultiQC/pull/1381#discussion_r660300270

I think for your use-case (each lane is being run exclusively), there's a sound argument as to why the undetermined reads should exist and will see what we can do. 

For our use case, we were not splitting by lane, but by sample type, since each sample type had various different bcl convert settings, such as adapter reads, override cycles and minimum trimmed read length.  There would often be situations where multiple bclconvert runs would process different samples on the same lane. This would mean the undetermined reads were abnormally high on both bclconvert runs, one could use some methods to try determine the number of actually undetermined reads by cross-referencing large undetermined counts with samples on other bcl-convert runs.  

From reading the comment above, the issue wasn't so much about the calculation, but ensuring that undetermined indices weren't merged over different sequencing runs? @andrei-seleznev am I right in paraphrasing that much? 

Is there a way to determine how many sequencing runs have been scanned by the multiqc as opposed to multiple bclconvert workflows over the same run?

**Comment by matthdsm** [comment]
Thanks for the comment! That does clarify the reasoning behind it.

I the case of multiple runs, I agree the undetermined runs should be hidden. They are, however, very useful in our case where we split each run by lane. This is also the default through basespace if I'm not mistaken.

**Comment by ewels** [comment]
> This is also the default through basespace if I'm not mistaken.

I'm tempted to follow this if this is the behaviour elsewhere.

@nvnieuwk - if you could you add a check for a MultiQC config parameter to switch this behaviour on and off, that would be great. Then people who want the current behaviour can keep it by setting a config flag.

See https://github.com/ewels/MultiQC/commit/0f04894d2e8f649679026c3655006a821fdd5929 for an implementation example.

Hopefully everyone will be happy then and we can merge? @alexiswl ?

**Comment by nvnieuwk** [comment]
The option has been added!

File: multiqc/modules/bclconvert/bclconvert.py [hunk 1]
```diff
@@ -182,7 +185,7 @@ def __init__(self):
         )
 
         # Add section with undetermined barcodes
-        if self.num_demux_files == 1:
+        if create_undetermined_barplots:
```
Comment (ewels) [inline multiqc/modules/bclconvert/bclconvert.py:188]: I think we still want to do it by default if only a single demux file.. I can't see a reason _not_ to do it in this case.

Comment (ewels) [reply multiqc/modules/bclconvert/bclconvert.py:189]: Addressed in  e9c0e26

**Review by ewels — APPROVED** [review]

**Comment by ewels** [comment]
Ok, thanks everyone! Hopefully this PR is a good compromise for all. Please shout if that's not the case.

**Pull request merged**